### PR TITLE
[Placement Group] Make placement group commit resource rpc request batched

### PR DESCRIPTION
**Pull request by clay4megtr**
## Why are these changes needed?

This is one part of this refactor, #20715 , make the commit resource RPC requests batched per node.

## Related issue number

#20715

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc [hunk 1]
```diff
@@ -339,26 +339,31 @@ void GcsPlacementGroupScheduler::CommitAllBundles(
     const std::shared_ptr<LeaseStatusTracker> &lease_status_tracker,
     const PGSchedulingFailureCallback &schedule_failure_handler,
     const PGSchedulingSuccessfulCallback &schedule_success_handler) {
-  const std::shared_ptr<BundleLocations> &prepared_bundle_locations =
-      lease_status_tracker->GetPreparedBundleLocations();
+  const std::unordered_map<NodeID,
+                           std::vector<std::shared_ptr<const BundleSpecification>>>
+      &prepared_bundle_locations =
+          lease_status_tracker->GetGroupedPreparedBundleLocations();
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc:345]:
```suggestion
          lease_status_tracker->GetPreparedBundleLocationsPerNode();
```

File: src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc [hunk 2]
```diff
@@ -808,6 +818,11 @@ const std::shared_ptr<BundleLocations> &LeaseStatusTracker::GetPreparedBundleLoc
   return preparing_bundle_locations_;
 }
 
+const std::unordered_map<NodeID, std::vector<std::shared_ptr<const BundleSpecification>>>
+    &LeaseStatusTracker::GetGroupedPreparedBundleLocations() const {
+  return grouped_preparing_bundle_locations_;
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc:823]: Why don't we just construct on the fly? In this way, we can avoid having a new separate field?

**Review by rkooo567 — COMMENTED** [review]
LGTM. Two small comments!

File: src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc (hunk 1 above)
Comment (clay4444) [reply src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc]: Removed.

File: src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc (hunk 2 above)
Comment (clay4444) [reply src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc]: Updated.

**Comment by rkooo567** [comment]
Failed tests seem unrelated

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Thanks for the great contribution! This is a great improvement, and it will help us optimizing one of N^2 bottlenecks.

**Pull request merged**